app/controllers/recipes_controller.rb

```diff
@@ -17,6 +20,18 @@ def new
   # GET /recipes/1/edit
   def edit; end
 
+  def toggle_privacy
+    @recipe = Recipe.find(params[:id])
+    @recipe.toggle_privacy!
+    redirect_to recipe_path(@recipe)
+  end
+
+  def toggle_shopping_tag
+    @recipe = Recipe.find(params[:id])
+    @recipe.toggle_shopping_tag!
+    redirect_to recipe_path(@recipe)
+  end
```

Comment:
Good implementation of the toggle functionality

Reply:
thanks